apps/meteor/app/api/server/api.ts

```diff
@@ -787,14 +788,14 @@ export class APIClass<TBasePath extends string = ''> extends Restivus {
 						} as unknown as SuccessResult<Record<string, any>>;
 					}
 
-					this.user = Meteor.users.findOne(
+					this.user = (await Users.findOne(
 						{
 							_id: auth.id,
 						},
 						{
 							fields: getDefaultUserFields(),
 						},
-					) as any;
+					)) as any;
 
 					this.userId = (this.user as unknown as IUser)?._id as any;
```

Comment:
Typing `this.user` would prevent this

Reply:
it's typed already, but the typings are really complex inside this internalRouterHandler function, there's a lot of contexts replacements 😞